src/drivers/general/queries/IrqlNotSaved/IrqlNotSaved.ql

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
Since this is a warning/low severity issue with only medium precision, I assert that this isn't quite ready to introduce into our collection yet. I recommend keeping it in the experimental directory until the full version is complete and the rule can detect if there are any code paths which don't save the IRQL.

If you open bugs with the rule as is, you have high chance of devs interpreting a duplicate bug when the rule is complete because it will cover the same code paths with inverse logic: one bug for not having a code path which saves, one bug for having a code path which doesn't save.

Reply:
FWIW, I selected "medium" here not because of false positives, but because of false negatives.  I expect this rule to miss (and it does miss) some cases due to the lack of must-flow analysis.  But I expect the false-positive rate to be low.  Do we want to use only false positive rate to determine if something is "experimental?"

You do make a good point about potential dev confusion if we upgrade the rule and file additional bugs.